File: regexlib/2021-5-15/python_re_test_file/regexlib_7740.py
# 7740
# [\\""=:;,](([\w][\w\-\.]*)\.)?([\w][\w\-]+)(\.([\w][\w\.]*))?\\sql\d{1,3}[\\""=:;,]
# POLYNOMIAL
# nums:4
# POLYNOMIAL AttackString:"\\"+"000."*5000+"!1 __POA(i)"

import re
from time import perf_counter

regex = """[\\""=:;,](([\w][\w\-\.]*)\.)?([\w][\w\-]+)(\.([\w][\w\.]*))?\\sql\d{1,3}[\\""=:;,]"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "\\" + "000." * i * 10000 + "!1 __POA(i)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/2021-5-15/javascript_test_file/regexlib_7740.js
/* 7740
 * [\\""=:;,](([\w][\w\-\.]*)\.)?([\w][\w\-]+)(\.([\w][\w\.]*))?\\sql\d{1,3}[\\""=:;,]
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:"\\"+"000."*5000+"!1 __POA(i)"
 */
var REGEX = /[\\""=:;,](([\w][\w\-\.]*)\.)?([\w][\w\-]+)(\.([\w][\w\.]*))?\\sql\d{1,3}[\\""=:;,]/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '\\' + '000.'.repeat(i*10000) + '!1 __POA(i)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}